src/integrations/cleanup-integration.php

```diff
@@ -440,6 +443,49 @@ protected function clean_indexables_for_non_publicly_viewable_taxonomies( $limit
 		// phpcs:enable
 	}
 
+	/**
+	 * Cleans up any indexables that belong to post type archive page that are not/no longer publicly viewable.
+	 *
+	 * @param int $limit The limit we'll apply to the queries.
+	 *
+	 * @return bool|int The number of deleted rows, false if the query fails.
+	 */
+	protected function clean_indexables_for_non_publicly_viewable_post_type_archive_pages( $limit ) {
+		global $wpdb;
+		$indexable_table = Model::get_table_name( 'Indexable' );
+
+		$included_post_types = $this->post_type->get_indexable_post_archives();
```

Comment:
This returns an array of objects.

Which makes the following query misbehave and delete ALL `post-type-archive`, while also throwing a notice